### PR TITLE
chore: use non-legacy Keras optimizers in examples and test fixtures [MLG-443]

### PR DESCRIPTION
**Pull request by drh-determined-ai**
## Description

Google-style imports; fix message in asserts about `request_stop_step_id`; undo some unwanted changes to ancient-checkpoints code.

Use `tensorflow.keras.optimizers` everywhere instead of `tensorflow.keras.optimizers.legacy`. This required some additional changes to how we do learning rate decay. Change from the deprecated kwarg `lr` to `learning_rate` (not doing this causes numerical errors).

## Test Plan

Automated tests.

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [x] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: examples/computer_vision/iris_tf_keras/model_def.py [hunk 1]
```diff
@@ -48,31 +43,33 @@ class IrisTrial(keras.TFKerasTrial):
...
-        model = self.context.wrap_model(Model(inputs=inputs, outputs=dense2))
+        model = self.context.wrap_model(models.Model(inputs=inputs, outputs=dense2))
 
         # Create and wrap the optimizer.
-        optimizer = RMSprop(
-            lr=self.context.get_hparam("learning_rate"),
-            decay=self.context.get_hparam("learning_rate_decay"),
+        lr_schedule = optimizers.schedules.ExponentialDecay(
+            initial_learning_rate=self.context.get_hparam("learning_rate"),
+            decay_steps=10000,
```
Comment (rb-determined-ai) [inline examples/computer_vision/iris_tf_keras/model_def.py:63]: where did this value come from?

File: harness/tests/experiment/fixtures/ancient-checkpoints/0.17.6-keras/code/model_def.py
Comment (rb-determined-ai) [inline harness/tests/experiment/fixtures/ancient-checkpoints/0.17.6-keras/code/model_def.py:1]: nothing in old checkpoint artifacts should be touched.

If you have to touch it, that means we've effectively failed the old checkpoint loading test.

I see that this file has been changed a couple of times.  Instead of updating it, can you revert it back to when it was originally added?

File: harness/tests/experiment/fixtures/ancient-checkpoints/0.17.7-keras/code/model_def.py
Comment (rb-determined-ai) [inline harness/tests/experiment/fixtures/ancient-checkpoints/0.17.7-keras/code/model_def.py:1]: ditto on this file, please revert changes you've made since it was first added.

File: examples/computer_vision/iris_tf_keras/model_def.py (hunk 1 above)
Comment (drh-determined-ai) [reply examples/computer_vision/iris_tf_keras/model_def.py:63]: It's a necessary argument. [Example code tends to just use a large-ish number](https://keras.io/api/optimizers/learning_rate_schedules/exponential_decay/).
I can take the magic number out of the Python code and put it in the experiment config along with the initial rate and decay.

**Review by rb-determined-ai — COMMENTED** [review]
Let's add a release note about what we expect end users to do to keep up with the change we made.

Please mention when the tf 2.11 image shipped, if it was before this version (which I think it was)

**Comment by drh-determined-ai** [comment]
> Let's add a release note about what we expect end users to do to keep up with the change we made.
> 
> Please mention when the tf 2.11 image shipped, if it was before this version (which I think it was)

There's no mention of `tf-2.11` in [bumpenvs.yaml v0.21.1](https://github.com/determined-ai/determined/blob/42ea168e861b6a02e2bebcd4a8c2ffeb359e073a/tools/scripts/bumpenvs.yaml), so I think we're ok.

**Review by rb-determined-ai — APPROVED** [review]
nice

**Comment by rb-determined-ai** [comment]
closing this for now, but will reopen in the future when we have bandwidth to figure out what went wrong
